lib/analyzer/os-release-detector.ts

```diff
@@ -35,7 +37,16 @@ async function detect(targetImage: string, options?: DockerOptions):
   }
 
   if (!osRelease) {
-    throw new Error('Failed to detect OS release');
+    if (envDetails.dockerFile &&
+      envDetails.dockerFile.baseImage === 'scratch') {
+
+      // If the docker file was build from a scratch image
+      // then we don't have a known OS
+
+      osRelease = { name: 'scratch', version: '0.0' };
```

Comment:
Added this as a special case when we have a scratch image with a bare bone environment.